src/input/manager.rs

```diff
@@ -1275,20 +1276,15 @@ impl Manager {
     }
 
     /// Returns the next available composite device dbus path
-    fn next_composite_dbus_path(&self) -> String {
-        let max = 2048;
-        let mut i = 0;
-        loop {
-            if i > max {
-                return "Devices exceeded".to_string();
-            }
+    fn next_composite_dbus_path(&self) -> Result<String, Box<dyn Error>> {
+        for i in 0u64.. {
```

Comment:
That's a lot of devices! :P

Reply:
> That's a lot of devices! :P

Indeed. This was done because udev has the device_num that is an u64, so I wanted to match that. It's more logical than a real problem to be fair, but it won't hurt.

Reply:
The old UdevDevice (I think it's called like this) wrongly uses u32 still and it parses that number from the output of a CLI command.

In general I think that old class has to be completely removed in favor of the new one that more than half software already uses.

This modification will make things easier when such modification is done (or the correct type is used and that struct is kept)